### DISCUSSION TITLE
Cache specific key only

### DISCUSSION THREAD

**Discussion by krisnapy** [vercel/swr#2980] [Q&A]
Is it possible to persistence cache just for a specific key path only?

**Comment by Weixuanf** [comment]
hi this is what im doing. it seems to work.

```
const apiKeysToCache = [
  "/api/workflow/listMyFlows",
  "/api/job/listMyJob",
];
export function swrLocalStorageProvider() {
  if (typeof window === "undefined") {
    // Return a dummy storage for server-side rendering
    return new Map();
  }
  // When initializing, we restore the data from `localStorage` into a map.
  const map = new Map<string, any>(
    JSON.parse(localStorage.getItem("app-cache") || "[]"),
  );

  // Before unloading the app, we write back all the data into `localStorage`.
  window.addEventListener("beforeunload", () => {
    const cacheMap = new Map<string, any>();
    for (const key of apiKeysToCache) {
      cacheMap.set(key, map.get(key));
    }
    const appCache = JSON.stringify(Array.from(cacheMap.entries()));
    localStorage.setItem("app-cache", appCache);
  });

  // We still use the map for write & read for performance.
  return map;
}

```

**Reply by cc7gs** [reply]
Here's how to store in asynchronous scenarios, such as in conjunction with localForage

**Comment by krisnapy** [comment]
> hi this is what im doing. it seems to work.
> 
> ```
> const apiKeysToCache = [
>   "/api/workflow/listMyFlows",
>   "/api/job/listMyJob",
> ];
> export function swrLocalStorageProvider() {
>   if (typeof window === "undefined") {
>     // Return a dummy storage for server-side rendering
>     return new Map();
>   }
>   // When initializing, we restore the data from `localStorage` into a map.
>   const map = new Map<string, any>(
>     JSON.parse(localStorage.getItem("app-cache") || "[]"),
>   );
> 
>   // Before unloading the app, we write back all the data into `localStorage`.
>   window.addEventListener("beforeunload", () => {
>     const cacheMap = new Map<string, any>();
>     for (const key of apiKeysToCache) {
>       cacheMap.set(key, map.get(key));
>     }
>     const appCache = JSON.stringify(Array.from(cacheMap.entries()));
>     localStorage.setItem("app-cache", appCache);
>   });
> 
>   // We still use the map for write & read for performance.
>   return map;
> }
> 
> ```

Thank you @Weixuanf 🙏🏼
